### ISSUE TITLE
GPU support for objective functions

### ISSUE THREAD

**Issue by bvanessen** [LLNL/lbann#346]

**Comment by DylanMcKinney** [comment]
I am interested in getting some more experience with GPU code. Taking a crack at this

**Comment by timmoon10** [comment]
CPU layers we should support:
- [x] Cross entropy
- [ ] Mean squared error
- [ ] Mean absolution deviation
- [ ] Binary cross entropy
- [ ] Softmax cross entropy
- [x] Sigmoid binary cross entropy

GPU layers we should support:
- [x] Cross entropy
- [ ] Mean squared error
- [ ] Mean absolution deviation
- [ ] Binary cross entropy
- [ ] Softmax cross entropy
- [ ] Sigmoid binary cross entropy
